python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/utils.py

```diff
@@ -119,10 +119,16 @@ def _get_p2p_data(self: AvdStructuredConfigCoreInterfacesAndL3EdgeProtocol, p2p_
         # Set descriptions or fallback to list with None values
         descriptions = p2p_link.descriptions or [None, None]
 
+        try:
+            ip_idx = ip[index]
+        except IndexError as exc:
+            msg = "IP_pools cannot allocate more than 2 ips but it can be done manually"
+            raise AristaAvdError(msg) from exc
+
```

Comment:
Hmm I think it is too late here - we should do the logic in the `_resolve_p2p_ips` function.
Though if we have more than 2 nodes the peer index is also broken sadly

@ClausHolbechArista I have seen people use this model with more then 2 nodes in a same link happily but I am wondering if we should remove this possibility in 6.0. We should really have had `max_length` on node to be 2 in the schema but I feel it is breaking


@Vibhu-gslab the error message should be that if you need 3 ips then you need to give them (and make sure all lists are the right size..)

```
p2p_links:
  - nodes: [a, b, c]
    ip: [IP_a, IP_b, IP_c]
```

in your input on the link

Reply:
@Vibhu-gslab after discussion with @ClausHolbechArista lets do the following:

1. the error message catching any `IndexError` (including the potential one for description line 135) should just output an error message stating `p2p_links model is intended to work for only two devices per entry.`)
2. Add a TODO in the schema for 6.0 stating we want to add a min_length of 2 and max_length of 2 to the lists (description, ip, nodes)

Reply:
group in the try/catch block the checks for ip, peer_ip and description

Rename `ip` to `ips` as really this is a list:

```
 # Set ips or fallback to list with None values
 ips = p2p_link.ip or [None, None]
```

and then 

```
        try:
            ip = ips[index]
            peer_ip = ips[peer_index]
            description = descriptions[index] 
        except IndexError as exc:
            msg = "IP_pools cannot allocate more than 2 ips but it can be done manually"
            raise AristaAvdError(msg) from exc
```

and use these variables in `data` below.

Reply:
Fixed!